test/cmd/core.sh

```diff
@@ -25,30 +25,30 @@ run_configmap_tests() {
   create_and_use_new_namespace
   kube::log::status "Testing configmaps"
   kubectl create -f test/fixtures/doc-yaml/user-guide/configmap/configmap.yaml
-  kube::test::get_object_assert 'configmap/test-configmap' "{{$id_field}}" 'test-configmap'
-  kubectl delete configmap test-configmap "${kube_flags[@]}"
+  kube::test::get_object_assert 'configmap/test-configmap' "{{${id_field:?}}}" 'test-configmap'
+  kubectl delete configmap test-configmap "${kube_flags[@]:?}"
 
   ### Create a new namespace
   # Pre-condition: the test-configmaps namespace does not exist
-  kube::test::get_object_assert 'namespaces' '{{range.items}}{{ if eq $id_field \"test-configmaps\" }}found{{end}}{{end}}:' ':'
+  kube::test::get_object_assert 'namespaces' '{{range.items}}{{ if eq '"$id_field"' \"test-configmaps\" }}found{{end}}{{end}}:' ':'
   # Command
   kubectl create namespace test-configmaps
   # Post-condition: namespace 'test-configmaps' is created.
   kube::test::get_object_assert 'namespaces/test-configmaps' "{{$id_field}}" 'test-configmaps'
 
   ### Create a generic configmap in a specific namespace
   # Pre-condition: configmap test-configmap and test-binary-configmap does not exist
-  kube::test::get_object_assert 'configmaps' '{{range.items}}{{ if eq $id_field \"test-configmap\" }}found{{end}}{{end}}:' ':'
-  kube::test::get_object_assert 'configmaps' '{{range.items}}{{ if eq $id_field \"test-binary-configmap\" }}found{{end}}{{end}}:' ':'
+  kube::test::get_object_assert 'configmaps' '{{range.items}}{{ if eq '"$id_field"' \"test-configmap\" }}found{{end}}{{end}}:' ':'
```

Comment:
this quoting looks suspect? why are we doing `'foo'"$bar"'baz'` ?

Reply:
yes, it's just to make `$bar` double quoted, another way is using double quote outside and escape `\"test-configmap\"` as `"{{range.items}}{{ if eq $id_field \\\"test-configmap\\\\" }}found{{end}}{{end}}:"`